packages/api-request/src/middlewares/test/nonce.js

```diff
@@ -0,0 +1,32 @@
+import createNonceMiddleware from '../nonce';
+
+describe( 'Nonce middleware', () => {
+	it( 'should add a nonce header to the request', () => {
+		const nonce = 'nonce';
+		const nonceMiddleware = createNonceMiddleware( nonce );
+		const requestOptions = {
+			method: 'GET',
+			path: '/wp/v2/posts',
+		};
+		const callback = ( options ) => {
+			expect( options.headers[ 'X-WP-Nonce' ] ).toBe( nonce );
+		};
+
+		nonceMiddleware( requestOptions, callback );
+	} );
+
+	it( 'should not add a nonce header to requests with nonces', () => {
+		const nonce = 'nonce';
+		const nonceMiddleware = createNonceMiddleware( nonce );
+		const requestOptions = {
+			method: 'GET',
+			path: '/wp/v2/posts',
+			headers: { 'X-WP-Nonce': 'existing nonce' },
+		};
+		const callback = ( options ) => {
+			expect( options.headers[ 'X-WP-Nonce' ] ).toBe( 'existing nonce' );
+		};
+
+		nonceMiddleware( requestOptions, callback );
```

Comment:
We should add [expect.hasAssertions](https://facebook.github.io/jest/docs/en/expect.html#expecthasassertions) call to both tests to make sure that `callback` was executed.